test/graphql/project_sample_query_test.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class ProjectSampleQueryTest < ActiveSupport::TestCase
+  PROJECT_QUERY = <<~GRAPHQL
+    query($projectPath: ID!) {
+      project(fullPath: $projectPath) {
+        name
+        path
+        id
+        samples {
+          nodes {
+            id
+            name
+            project {
+              id
+            }
+          }
+        }
+      }
+    }
+  GRAPHQL
+
+  def setup
+    @user = users(:john_doe)
+  end
+
+  test 'project with sample query should work' do
+    project = projects(:project1)
+
+    result = IridaSchema.execute(PROJECT_QUERY, context: { current_user: @user },
+                                                variables: { projectPath: project.full_path })
+
+    assert_nil result['errors'], 'should work and have no errors.'
+
+    data = result['data']['project']
+
+    assert_not_empty data, 'project type should work'
+    assert_equal project.name, data['name']
+
+    assert_equal project.to_global_id.to_s, data['id'], 'id should be GlobalID'
+
+    # verify fetched sample data matches data on project
+    project.samples.each_with_index do |sample, index|
```

Comment:
**nitpick (blocking):** Adding `first` above this will need to be changed to just ensure that the samples are present in the project. So If you use `first: 10` just grab the first 10 samples from the project and ensure they match here. You can also check `totalCount` and see that it matches the number of samples that the project has i.e. `Project.samples.count`

Reply:
Fixed.